server/src/exporter.js

```diff
@@ -0,0 +1,255 @@
+
+function exportPath(deviceId) {
+  if (deviceId.search(/^[a-z0-9\-]+$/i) == -1) {
+    reject(new Error("Bad deviceId"));
+  }
+  return path.join(config.exportBase, "export---" + deviceId);
+}
+
+function removeExportPath(dir) {
+  return new Promise((resolve, reject) => {
+    rimraf(dir, (error) => {
+      if (error && error.code != "ENOENT") {
+        reject(error);
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+function launchWget(deviceId) {
+  return new Promise((resolve, reject) => {
+    let dir = exportPath(deviceId);
+    let script = path.join(__dirname, "export-shots.sh");
+    let start = Date.now();
+    let subprocess = child_process.spawn(
+      script,
+      {
+        env: {
+          BASE_DIR: dir,
+          AUTH: require("./dbschema").getTextKeys()[0],
+          DEVICE_ID: deviceId,
+          PORT: config.port,
+          CONTENT_PORT: config.contentPort,
+        }
+      }
+    );
+    subprocess.on("close", function (code) {
+      console.info("Exported to", dir, "in", (Date.now() - start) + "ms", "exit code:", code);
+    });
+    subprocess.on("error", function (e) {
+      console.warn("Error launching wget:", e);
+    });
+    resolve();
+    // FIXME: I don't think an exception here will reject
+    // Also maybe this is all synchronous
+  });
+}
+
+function getWgetStatus(deviceId) {
+  return new Promise((resolve, reject) => {
+    let dir = exportPath(deviceId);
+    let result = {errors: []};
+    let steps = 3;
+    function done() {
+      if (steps <= 0) {
+        resolve(result);
+      }
+    }
+    fs.readFile(path.join(dir, "wget-process.pid"), {encoding: "UTF-8"}, (error, data) => {
+      if (error && error.code == "ENOENT") {
+        // File doesn't exist
+        result.running = false;
+      } else if (error) {
+        result.errors.push(["Error loading process PID:", error]);
+      } else {
+        result.running = true;
+      }
+      steps--;
+      done();
+    });
+    fs.readFile(path.join(dir, "script-exit-code.txt"), {encoding: "UTF-8"}, (error, data) => {
+      if (error && error.code == "ENOENT") {
+        // Do nothing
+      } else if (error) {
+        result.errors.push(["Error loading exit code:", error]);
+      } else {
+        let code = parseInt(data, 10);
+        if (code) {
+          result.success = false;
+        } else {
+          result.success = true;
+        }
+      }
+      steps--;
+      done();
+    });
+    fs.stat(path.join(dir, "pageshot-export.zip"), (error, stat) => {
+      if (error && error.code == "ENOENT") {
+        result.zip = false;
+      } else if (error) {
+        result.errors.push(["Error checking zip file:", error]);
+      } else {
+        result.zip = {
+          size: stat.size,
+          mtime: stat.mtime.getTime(),
+          keepTime: keepTime
+        };
+      }
+      steps--;
+      done();
+    });
+  });
+}
+
+exports.cleanExports = function () {
```

Comment:
It occurred to me this function might be simpler if it called `getWgetStatus()`

Also maybe it should be a script and not written in Node, making it wildly simpler.